module/mempool/herocache/backdata/heropool/pool_test.go

```diff
@@ -374,52 +373,49 @@ func testInvalidatingTail(t *testing.T, pool *Pool, entities []*unittest.MockEnt
 			//
 			// used tail should move backward after each invalidation
 			require.Equal(t, entities[size-i-2].ID(), usedTail.id)
-			require.Equal(t, EIndex(size-i-2), pool.used.tail.getSliceIndex())
+			require.Equal(t, EIndex(size-i-2), pool.used.tail)
 
 			// used head must point to the first element in the pool,
 			require.Equal(t, entities[0].ID(), usedHead.id)
-			require.Equal(t, EIndex(0), pool.used.head.getSliceIndex())
+			require.Equal(t, EIndex(0), pool.used.head)
 		} else {
 			// pool is empty
 			// used head and tail must be nil and their corresponding
 			// pointer indices must be undefined.
 			require.Nil(t, usedHead)
 			require.Nil(t, usedTail)
-			require.True(t, pool.used.tail.isUndefined())
-			require.True(t, pool.used.head.isUndefined())
+			require.True(t, pool.used.size == 0)
+			//require.True(t, pool.used.head.isUndefined())
 		}
 	}
 }
 
 // testInitialization evaluates the state of an initialized pool before adding any element to it.
 func testInitialization(t *testing.T, pool *Pool, _ []*unittest.MockEntity) {
-	// head and tail of "used" linked-list must be undefined at initialization time, since we have no elements in the list.
-	require.True(t, pool.used.head.isUndefined())
-	require.True(t, pool.used.tail.isUndefined())
```

Comment:
Shoudn't these two conditions be like the following? 
```go
			require.Zero(t, pool.used.tail)
			require.Zero(t, pool.used.head)
```

Reply:
Addressed in comments below.